src/test/java/org/folio/rest/api/HoldingsStorageTest.java

```diff
@@ -603,6 +610,60 @@ public void cannotCreateHoldingWithoutPermanentLocation() throws Exception {
       is("permanentLocationId"));
   }
 
+  /**
+   * Create three instances, and for each of them return a JsonObject of a new holding belonging to it.
+   * The holdings are not created.
+   */
+  private JsonArray threeHoldings() {
+    JsonArray holdingsArray = new JsonArray();
+    for (int i=0; i<3; i++) {
+      UUID instanceId = UUID.randomUUID();
+      try {
+        instancesClient.create(smallAngryPlanet(instanceId));
+      } catch (MalformedURLException | InterruptedException | ExecutionException | TimeoutException e) {
+        throw new RuntimeException(e);
+      }
+      holdingsArray.add(new JsonObject()
+          .put("id", UUID.randomUUID().toString())
+          .put("instanceId", instanceId.toString())
+          .put("permanentLocationId", mainLibraryLocationId.toString()));
+    }
+    return holdingsArray;
+  }
+
+  private Response postSync(JsonArray holdingsArray) {
+    JsonObject holdingsCollection = new JsonObject().put("holdingsRecords", holdingsArray);
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+    client.post(holdingsStorageSyncUrl(""), holdingsCollection, TENANT_ID, ResponseHandler.any(createCompleted));
+    try {
+      return createCompleted.get(5, SECONDS);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void canSyncPost() {
+    JsonArray holdingsArray = threeHoldings();
+    assertThat(postSync(holdingsArray), statusCodeIs(HttpStatus.HTTP_CREATED));
+    for (Object holding : holdingsArray) {
+      assertExists((JsonObject) holding);
+    }
+  }
+
+  @Test
+  public void cannoteSyncPostWithDuplicateId() {
```

Comment:
I think it could be valuable to also have a test where a batch is submitted with a record ID that already exists in the system

Reply:
Yes, I've added an test to InstanceStorageTest, now there are three kinds of failing tests:
* cannotPostSynchronousBatchWithInvalidInstance
* cannotPostSynchronousBatchWithExistingId
* cannotPostSynchronousBatchWithDuplicateId
For Holdings and Items I only have a single failing test because it is the same StorageHelper.postSync code, no need to duplicate the tests.

Reply:
> For Holdings and Items I only have a single failing test because it is the same StorageHelper.postSync code, no need to duplicate the tests.

I understand this trade off. In general, I'm not keen on relying on implementation similarity for API tests, which I think are intended to be as ignorant of that as possible and act as documentation for API usage.